### PR TITLE
[14.0][ADD] new module: project_forecast_line_priority

### PR DESCRIPTION
**Pull request by ntsirintanis**
Related and based on https://github.com/OCA/project/pull/1238

### CODE REVIEW

**Review by gjotten — APPROVED** [review]
:+1: functional review

**Comment by ntsirintanis** [comment]
@NL66278 @thomaspaulb please review and let's try to get this merged

File: project_forecast_line_priority/models/project_task.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    def _get_forecast_date_planned(self, priority=None):
+        """Update forecast date end based on priority"""
+        self.ensure_one()
+        if self.date_deadline:
+            return False
+        priority = priority or self.priority
+        if not priority:
+            # This may happen when a portal user
+            # is creating a task on portal
+            priority = "0"
```
Comment (thomaspaulb) [inline project_forecast_line_priority/models/project_task.py]: Why not `priority = priority or self.priority or "0"`

File: project_forecast_line_priority/models/project_task.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        selection = self.company_id["forecast_line_priority_%s_selection" % priority]
+        if selection == "delta":
+            return fields.Date.today() + timedelta(
+                days=int(self.company_id["forecast_line_priority_%s_delta" % priority])
+            )
+        if selection == "date":
+            return self.company_id["forecast_line_priority_%s_date" % priority]
```
Comment (thomaspaulb) [inline project_forecast_line_priority/models/project_task.py]: In the rare case where there's some module adding more priority possibilities or something, we can add some extra checking with:

```
return getattr(self.company_id, "forecast_line_priority_%s_date" % priority, None)
```

or

```
field_name = "forecast_line_priority_%s_date" % priority
if hasattr(self.company_id, field_name):
    return self.company_id[field_name]
else
    return 'something_else'
```

**Review by thomaspaulb — APPROVED** [review]
non blocking code review

**Comment by gjotten** [comment]
@thomaspaulb if you sign off on this we can get it merged :+1:

**Review by thomaspaulb — APPROVED** [review]

**Comment by thomaspaulb** [comment]
@OCA/project-service-maintainers merge?

**Pull request merged**